### PR TITLE
Views

### PR DESCRIPTION
**Pull request by danielsafari143**
In this pull request, I have added all of the views for my blog application.

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,13 @@
...
+          <li>
+            <p>
+                <img src=<%=item.photo%> alt='Profile'/>
+                UserName :  <%=item.name%>
+            </p>
+            <p>Number of posts: <%= item.posts_counter%></p>
+          </li>
+        <% end %>
+    </ul>
+</div>
```
Comment (KanzaTahreem) [inline app/views/users/index.html.erb]: - [x] You have done a good job displaying a list of users. However, please ensure that the index view page matches the provided [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_root_users.png) ✨Kindly take a look at the video below. When I use a correct URL for an image, this is how it renders 👇 Please make sure to follow the provided wireframe for the other elements of the index page

https://github.com/danielsafari143/blog-app/assets/106119105/64b32780-a2ae-4b17-9320-d9f559430e73


- [x] Kindly analyze the wireframes deeply and try to identify how a user would navigate from one page to another. For example, clicking on each user takes you to their show page as shown in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png). Currently, there's no method to navigate to the user's show page except by manually changing the URL, which not only provides a good user experience but also deviates from the project's requirements.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,19 @@
...
+    <ul>
+        <% @user.posts.each do |item| %>
+          <li><%= item.text %></li>
+          <p>
+            <span>comments: <%= item.comments_counter %></li> </span>, 
+            <span>Likes: <%= item.likes_counter %> </span>
+          </p>
+        <% end %>
+    </ul>
+</div>
```
Comment (KanzaTahreem) [inline app/views/users/show.html.erb]: - [x]  For the user's show page, please ensure that you follow the provided wireframe for styling your app.
- [x] Please make sure to utilize the `#three_recent_posts` method present in the user model to display the user's three most recent posts, as specified in the given wireframe. Currently, all of the user's posts are being displayed on the user's show page.
- [x] Here again, please ensure that **See all Post** links opens user's post index as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)
- [x] Kindly ensure that the post's title (for example, if it's post 1, post 2 etc) is also included as it is currently not visible. Additionally, please update the user's information displayed on the show page in accordance with the wireframe.

Current | Wireframe
---| ---|
image | image

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -1 +1,13 @@
...
+        <p>Post #<%=index%> </p>
+        <div>
+            <p><%= item.text %></p>
+             <% item.comments.each do |item| %>
+               <p>Comments : <%= User.find_by(id:item.author_id).name %> </p>
+             <% end %>
+        </div>
+        </span> Comment : <%= item.comments_counter %>, Likes: <%= item.likes_counter%> </span>
+    <% end %>
+</div>
```
Comment (KanzaTahreem) [inline app/views/posts/index.html.erb]: - [x] Please ensure that you follow the provided [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_all_posts.png) for styling the post index page
- [x] Please msure to use the `five_recent_comments` method present in the post model to render the 5 most recent comments on the `app/views/posts/index.html.erb` page, as highlighted in the wireframe. Currently, all comments from all posts are being rendered.
- [x]  Similarly, please ensure that there is a way to navigate to the post's show page. For instance, clicking on each post takes you to the post's show page. Currently, I couldn't find any way to navigate except by manually changing the URL. Please fix this for the reasons mentioned in the previous comment 
- [x] Also, please ensure that the comment's text and user information is displayed. Currently, only the user's name is visible. For comment's text you can use something like this 👇

```ruby
<p><%= comment.author.name %>: <%= comment.text %></p>
```

Current | Wireframe
---| ---|
image | image

File: app/views/posts/show.html.erb [hunk 4]
```diff
@@ -1 +1,13 @@
...
+    <div>
+        </span>Comment : <%= @post.comments_counter %>, Likes: <%= @post.likes_counter%></span>
+       <p> <%= @post.text %> </p>
+    </div>
+    <ul>
+        <% @comment.each do |item| %>
+            <p>Comments : <%= User.find_by(id:item.author_id).name %> </p>
+        <% end %>
+    </ul>
+</div>
```
Comment (KanzaTahreem) [inline app/views/posts/show.html.erb]: - [x] Please make sure to follow the provided [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_single_post.png) for styling your app
- [x] As mentioned above ☝, please ensure that the comment's text is displayed. Currently, only the user's name is visible.

Current | Wireframe
---| ---|
image | image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @danielsafari143 

Good job so far 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ `UsersController` is updated
- ✅ `PostsController` is updated
- ✅ Descriptive PR and README file
- ✅ Correct Linters

## Required Changes 🛑

- [x] The `issues page` link in contributing section is broken ⚠ Kindly make sure to fix this for maintaining a professional README

  image

- [x] Please ensure that we receive the user's index page when the app starts. Currently, Rails renders the default page. Please refer to the URL in the wireframe below 👇. You can achieve this by setting `root users#index` in `routes.rb`.

  image

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @danielsafari143, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested by the previous review (✅)) 👏
- Added the views associated with the controllers👏


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**